### PR TITLE
Code Quality: Merge P/Invoke into single partial class

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

Merged:
- FileUtils
- InteropHelpers
- NativeDirectoryChangesHelper

## PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
- [ ] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   _None_

## Screenshots

_None_

### CODE REVIEW

File: src/Files.App/Helpers/Win32/Win32Helper.Process.cs [hunk 1]
```diff
@@ -37,5 +37,76 @@ public static async Task<bool> InvokeWin32ComponentsAsync(IEnumerable<string> ap
...
+		/// <param name="path">Path of the file.</param>
+		/// <returns>Processes locking the file.</returns>
+		/// <remarks>
+		/// For more info, visit
+		/// <br/>
+		/// - <a href="https://learn.microsoft.com/ja-jp/windows/win32/api/restartmanager/nf-restartmanager-rmgetlist"/>
+		/// <br/>
+		/// - <a href="https://stackoverflow.com/questions/317071/how-do-i-find-out-which-process-is-locking-a-file-using-net/317209#317209"/>
+		/// </remarks>
+		public static List<Process> WhoIsLocking(string[] resources)
```
Comment (0x5bfa) [inline src/Files.App/Helpers/Win32/Win32Helper.Process.cs:53]: This has been inside of FileUtils, which has been interop helper.
So, this should've been inside of a normal helper.

File: src/Files.App/Helpers/Win32/Win32Helper.WindowManagement.cs [hunk 2]
```diff
@@ -37,5 +42,38 @@ public static unsafe void BringToForegroundEx(Windows.Win32.Foundation.HWND hWnd
...
 			PInvoke.AttachThreadInput(dwCurID, dwMyID, false);
 		}
+
+		/// <summary>
+		/// Sets cursor when hovering on a specific element.
+		/// </summary>
+		/// <param name="uiElement">An element to be changed.</param>
+		/// <param name="cursor">Cursor to change.</param>
+		public static void ChangeCursor(this UIElement uiElement, InputCursor cursor)
+		{
```
Comment (0x5bfa) [inline src/Files.App/Helpers/Win32/Win32Helper.WindowManagement.cs:52]: As above.

File: src/Files.App/Helpers/Win32/Win32Helper.WindowManagement.cs [hunk 3]
```diff
@@ -37,5 +42,38 @@ public static unsafe void BringToForegroundEx(Windows.Win32.Foundation.HWND hWnd
...
+		}
+
+		/// <summary>
+		/// Changes an attribute of the specified window.
+		/// </summary>
+		/// <param name="hWnd">A handle to the window and, indirectly, the class to which the window belongs.</param>
+		/// <param name="nIndex">The zero-based offset to the value to be set.</param>
+		/// <param name="dwNewLong">The replacement value.</param>
+		/// <returns>If the function succeeds, the return value is the previous value of the specified offset.</returns>
+		public static IntPtr SetWindowLong(HWND hWnd, WindowLongFlags nIndex, IntPtr dwNewLong)
```
Comment (0x5bfa) [inline src/Files.App/Helpers/Win32/Win32Helper.WindowManagement.cs:71]: As above.

**Review by 0x5bfa — COMMENTED** [review]
Explained some weird code changes.

File: src/Files.App/Data/Contracts/IDataTransferManagerInterop.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using System.Runtime.InteropServices;
+
+namespace Files.App.Data.Contracts
+{
```
Comment (0x5bfa) [inline src/Files.App/Data/Contracts/IDataTransferManagerInterop.cs:7]: As above.

File: src/Files.App/Data/Contracts/IDataTransferManagerInterop.cs
Comment (0x5bfa) [inline src/Files.App/Data/Contracts/IDataTransferManagerInterop.cs:1]: As above.

**Review by yaira2 — APPROVED** [review]

**Comment by yaira2** [comment]
How were each of these methods tested to ensure everything is functioning as before?

**Comment by 0x5bfa** [comment]
Obviously ensured safely because I just moved and added comments.

**Pull request merged**